hooks/advocate-conversion-utils.ts

```diff
@@ -42,38 +42,46 @@ class AdvocatesAirtableRecords extends AirtableRecords {
       fields: Object.values(this.recordFields),
       filterByFormula: `AND({${slackId}})`,
       sort: [{ field: this.recordFields.name, direction: 'asc' }]
-    }).eachPage((records, nextPage) => {
+    }).eachPage(async (records, nextPage) => {
       for (const record of records) {
+        this.id = record.id
         const advocate = this.convertToAdvocate(record)
-        advocates.push(advocate)
+        advocates.push(await advocate)
       }
       nextPage()
     })
     return Promise.resolve(advocates)
   }
 
-  convertToAdvocate (record: any): Advocate {
-    return {
+  async convertToAdvocate (record: any): Promise<Advocate> {
+    const advocate = {
       name: this.getName(record),
-      image: this.getImage(record),
+      image: await this.getImage(record),
       region: this.getRegion(record),
       city: this.getCity(record),
       country: this.getCountry(record),
       slackId: this.getSlackId(record),
       slackUsername: this.getSlackUsername(record)
     }
+    return advocate
   }
 
   public getName (record: any): string {
     return record.get(this.recordFields!.name)
   }
 
-  public getImage (record: any): string {
+  public async getImage (record: any): Promise<string> {
```

Comment:
We have two `getImage` functions, one for advocates and one for events. They only differ on the URLs, so I think we can create a reusable function passing to it the two URLs we need

Thinking in loud now: in addition, does it make sense to have functions like `getName`, where we only access an object property? if we want a function for that, we could create something like:

```ts
public getElementFromRecord (element:string, record: any): any {
    return record.get(this.recordFields![element])
  }
```

Reply:
As discussed offline, I like these improvement ideas. We'll follow up on https://github.com/Qiskit/qiskit.org/issues/2934 to avoid extending the scope of this pull request.